Tela de cadastro e de completar cadastro

gabrielsoaresm94 opened the pull request:
# ✏📝 Descreva as mudanças que esse Pull Request implementa
### Descreva aqui

## O que esse PR representa?
- [ ] Bug fix(mudança ininterrupta que corrige um problema) 🐞
- [x] New Feature/Novo recurso (mudança ininterrupta que adiciona funcionalidade) ⚙
- [ ] Mudança de última hora (correção ou recurso que faria com que a funcionalidade existente não funcionasse conforme o esperado) ⏰
- [ ] Alteração de documentação 📝

# Checklist:

- [ ] Adicionei testes que comprovam que minha correção é eficaz ou que meu recurso funciona 🧪
- [ ] Testes de unidade novos e existentes são aprovados localmente com minhas alterações 🧪
- [x] Meu código segue as diretrizes de estilo deste projeto 📏
- [ ] Realizei uma auto-revisão do meu próprio código 👨‍🏫👩‍🏫
- [ ] Removi arquivos inutilizados ♻
- [x] Fiz as alterações correspondentes com a minha task 🙏
- [x] Minhas alterações não geram novos problemas 🙏

gabrielsoaresm94 committed:
Cadastro simples adicionado

gabrielsoaresm94 committed:
Finalização do cadastro, ainda não adaptada para o mobile

brunodelucasbarbosa commented on src/components/CadastroForm/index.jsx:
```diff
@@ -0,0 +1,257 @@
+import AuthContext from '../../context/Auth/AuthProvider'
+import Button from '../Button'
+
+const CADASTRO_URL = 'https://aux-bolsistas-dev.herokuapp.com/cadastro'
```
Link errado
se já tá colocando em uma variavel, coloca em .env

gabrielsoaresm94 replied:
Troquei para `https://aux-bolsistas-dev.herokuapp.com//v1/students`.

brunodelucasbarbosa commented on src/components/CadastroForm/index.jsx:
```diff
@@ -0,0 +1,257 @@
+      >
+        CPF
+        <input
+          type="number"
```
validar o campo

brunodelucasbarbosa commented on src/components/CadastroForm/index.jsx:
```diff
@@ -0,0 +1,257 @@
+        Contato
+        <div className="flex flex-1">
+          <input
+            type="text"
```
validar o campo

brunodelucasbarbosa commented on src/components/CadastroForm/index.jsx:
```diff
@@ -0,0 +1,257 @@
+          />
+          <input
+            type="number"
+            id="phone_id"
```
validar o campo

brunodelucasbarbosa commented on src/components/CadastroForm/index.jsx:
```diff
@@ -0,0 +1,257 @@
+        htmlFor="confirm"
+        className="relative flex h-[89px] flex-col gap-y-0.5 text-base font-medium leading-7 text-gray-800"
+      >
+        Confirmar senha
```
Se as senhas são diferentes, deveria dar erro, não?

gabrielsoaresm94 replied:
Adicionado um if para caso forem diferentes chama um alerta.

brunodelucasbarbosa commented on src/components/CompleteCadastroForm/index.jsx:
```diff
@@ -0,0 +1,252 @@
+      >
+        Número matrícula
+        <input
+          type="number"
```
validar o campo

brunodelucasbarbosa commented on src/components/CompleteCadastroForm/index.jsx:
```diff
@@ -0,0 +1,252 @@
+        htmlFor="modalidade_id"
+        className="flex h-[109px] w-1/3 flex-col gap-y-1.5 px-2 text-base font-medium leading-7 text-gray-800"
+      >
+        Modalidade
```
Isso será alterado

gabrielsoaresm94 replied:
Alterei a modalidade para umas informações fixas, com o que achei no site das modalidades de bolsas de pesquisa. Podemos alterar depois, se precisar.

brunodelucasbarbosa commented on src/components/CompleteCadastroForm/index.jsx:
```diff
@@ -0,0 +1,252 @@
+        Nome do orientador
+        <select
+          id="orientador_id"
+          placeholder="Digite seu número de matrícula"
```
isso aqui acho que está errado

brunodelucasbarbosa commented on src/components/CompleteCadastroForm/index.jsx:
```diff
@@ -0,0 +1,252 @@
+      >
+        Link currículo Lattes
+        <input
+          type="number"
```
validar campo

brunodelucasbarbosa commented on src/components/CompleteCadastroForm/index.jsx:
```diff
@@ -0,0 +1,252 @@
+        Agência
+        <select
+          id="agencia_id"
+          placeholder="Digite seu número de matrícula"
```
....

brunodelucasbarbosa commented on src/components/CompleteCadastroForm/index.jsx:
```diff
@@ -0,0 +1,252 @@
+      >
+        Valor da bolsa
+        <input
+          type="number"
```
dá pra colocar máscara nisso pra facilitar

brunodelucasbarbosa commented on src/pages/Cadastro/index.jsx:
```diff
@@ -0,0 +1,22 @@
+        <p className="text-center text-base font-normal leading-6">
+          Já tem uma conta?{' '}
+          <a href="/#" className="text-blue-600 transition-colors hover:text-blue-800">
+            Entrar
```
Isso leva para a home, mas, a home não trás de volta.

gabrielsoaresm94 replied:
Troquei para levar para `/` e passei a utilizar o componente Link.

brunodelucasbarbosa replied:
Ao clicar em cadastre-se, o usuário tem que ser direcionado para a página de cadastro
![image](https://user-images.githubusercontent.com/60590932/194450189-8064d850-eb1a-4189-b26b-103256037899.png)

brunodelucasbarbosa commented on src/components/CadastroForm/index.jsx:
```diff
@@ -0,0 +1,257 @@
+    return '/'
+  }
+
+  async function onSubmit(data) {
```
Você tá utilizando estado no context pra guardar os estados, mas, isso vai se perder quando direcionar....

gabrielsoaresm94 replied:
Mudei para o padrão de Monique

brunodelucasbarbosa commented on src/components/CadastroForm/index.jsx:
```diff
@@ -0,0 +1,257 @@
+  }
+
+  async function onSubmit(data) {
+    try {
```
não está integrado

brunodelucasbarbosa reviewed (changes requested):
nao esta integrado

gabrielsoaresm94 committed:
remove error patterns

gabrielsoaresm94 committed:
Merge branch 'main' of https://github.com/SAB-UFBA2022/frontend into feature/cadastro

gabrielsoaresm94 committed:
fix: passa a utilizar os novos inputs para a tela de cadastro

gabrielsoaresm94 committed:
fix: atualiza pre-cadastro e adiciona novas funções

gabrielsoaresm94 committed:
fix: atualização de inputs em completar-cadastro

gabrielsoaresm94 committed:
feat: cria e substitui select por FormSelect

gabrielsoaresm94 committed:
fix: integra formulário com a api, pela primeira vez

gabrielsoaresm94 committed:
Merge branch 'main' of https://github.com/SAB-UFBA2022/frontend into feature/cadastro

gabrielsoaresm94 committed:
fix: remove logs

brunodelucasbarbosa commented on .env.example:
```diff
@@ -0,0 +1,2 @@
+BASE_URL=
+CADASTRO_URL=
```
n precisa desse cadastro url, só o base e você complementa com o path

brunodelucasbarbosa commented on src/App.js:
```diff
@@ -16,6 +18,8 @@ export default function App() {
         {/* Rotas públicas */}
         <Route path="/" element={<Login />} />
+        <Route path="/cadastro" element={<Cadastro />} />
+        <Route path="/complete-cadastro" element={<CompleteCadastro />} />
```
se o usuário acessar essa rota ele vai ter acesso ao `CompleteCadastro`, verificou se isso trás problemas? pois a requisição

gabrielsoaresm94 replied:
se o usuário não passar por cadastro antes, vai quebrar

brunodelucasbarbosa commented on src/components/CadastroForm/index.jsx:
```diff
@@ -0,0 +1,143 @@
+      displayAlert()
+    }
+    const dataUser = { name_id, tax_id, email_id, phone_id, password }
+    preSaveUser(dataUser)
```
Você usou LocalStorage, ok. 
`prevSaveUser `não está direcionando o usuário para a segunda rota.

gabrielsoaresm94 replied:
Sim, tem um problema no if do método, mas não consegui identificar ainda. Se puderem me ajudar com com useEffect e redirecionamento, iria ser mt bom.

brunodelucasbarbosa replied:
Vc n precisa usar useEffect, basta colocar no submit do form se estiver tudo okay, você redireciona....

gabrielsoaresm94 replied:
Beleza

brunodelucasbarbosa commented on src/components/CadastroForm/index.jsx:
```diff
@@ -0,0 +1,143 @@
+        Contato
+        <div className="flex flex-1">
+          <FormInput
+            type="text"
```
coloca isso tipo email, já que é email

gabrielsoaresm94 replied:
ok

brunodelucasbarbosa commented on src/components/CompleteCadastroForm/index.jsx:
```diff
@@ -0,0 +1,207 @@
+      !fim_id ||
+      !defesa_id
+    ) {
+      displayAlert()
```
essa função não tem um `return,` então ainda mostra que a primeira etapa deu certo
![image](https://user-images.githubusercontent.com/60590932/194448109-ad5b1b98-7557-497d-ae7d-1163d01797b3.png)

gabrielsoaresm94 replied:
Beleza, vou adicionar retorno que falta.

brunodelucasbarbosa commented on src/components/CompleteCadastroForm/index.jsx:
```diff
@@ -0,0 +1,207 @@
+      enrollment_date_pgcomp: defesa_id,
+      phone_number: userphone,
+      password: userpassword,
+      role: 'studant'
```
aqui deveria ser
`role: 'STUDENT'`

gabrielsoaresm94 replied:
Beleza, vou corrigir.

brunodelucasbarbosa commented on src/components/CompleteCadastroForm/index.jsx:
```diff
@@ -0,0 +1,207 @@
+          label="Nome do orientador"
+          id="orientador_id"
+          name="orientador_id"
+          value={{
```
Esse value em teste, já temos orientadores, você deve puxar do back-end os orientadores para listar aqui

gabrielsoaresm94 replied:
Ainda não sei como fazer, pretendo usar um método pra fazer a requisição GET e provavelmente usar o useEffect pra chamar esse método. Mas não sei como definir, quando a página é acessada.

brunodelucasbarbosa replied:
Procura pelo `useEffect `sem parâmetro https://medium.com/reactbrasil/entenda-como-funciona-o-useeffect-e743b86f104
Tu busca os orientadores, e mapeia eles, como você fez com o curso

gabrielsoaresm94 replied:
Adicionei o nome do professor, para funcionar como teste. Depois quando o POST tiver ok, posso investir nesse GET.

brunodelucasbarbosa commented on src/components/CompleteCadastroForm/index.jsx:
```diff
@@ -0,0 +1,207 @@
+      <div className="flex h-[109px] w-1/3 flex-col gap-y-1.5 px-2 text-base font-medium leading-7 text-gray-800">
+        <FormInput
+          label="Link currículo Lattes"
+          type="text"
```
aqui poderia ser url

gabrielsoaresm94 replied:
ok

brunodelucasbarbosa commented on src/components/CompleteCadastroForm/index.jsx:
```diff
@@ -0,0 +1,207 @@
+          label="Agência"
+          id="agencia_id"
+          name="agencia_id"
+          value={{
```
isso aqui vai ter que ser mudado também

brunodelucasbarbosa replied:
está usando value de teste, deveria puxar as agências do back

gabrielsoaresm94 replied:
Mesma resposta de cima, para orientador.

gabrielsoaresm94 replied:
Conversei com Lucas e resolvi deixar algumas agências como "hardcoded", por enquanto.

brunodelucasbarbosa commented on src/components/CompleteCadastroForm/index.jsx:
```diff
@@ -0,0 +1,207 @@
+          type="text"
+          id="bolsa_id"
+          name="bolsa_id"
+          value={values.bolsa_id}
```
Isso aqui futuramente vai ter que ser o valor em centavos por causa do padrão do backend

brunodelucasbarbosa replied:
Não necessariamente precisa mudar agora, só colocar um TODO para futuramente ser visto essa questão

gabrielsoaresm94 replied:
ok

brunodelucasbarbosa commented on src/components/CompleteCadastroForm/index.jsx:
```diff
@@ -0,0 +1,207 @@
+          type="date"
+          id="inicio_id"
+          name="inicio_id"
+          value={values.inicio_id}
```
Tu declarou esse valor mas não passou pra sua função, ou seja, esse campo não existe?

gabrielsoaresm94 replied:
Não passei pq não encontrei esse campo para usar na requisição do backend, avisei isso no Discord, acho pode ter campos faltando.

gabrielsoaresm94 replied:
Adicionado

brunodelucasbarbosa commented on src/components/CompleteCadastroForm/index.jsx:
```diff
@@ -0,0 +1,207 @@
+          type="date"
+          id="fim_id"
+          name="fim_id"
+          value={values.fim_id}
```
Tu declarou esse valor mas não passou pra sua função, ou seja, esse campo não existe?²

gabrielsoaresm94 replied:
Não passei pq não encontrei esse campo para usar na requisição do backend, avisei isso no Discord, acho pode ter campos faltando.²

brunodelucasbarbosa replied:
![image](https://user-images.githubusercontent.com/60590932/194548133-84ab4f08-5357-4699-9615-a3a26a26d3ae.png)
os campos existem

gabrielsoaresm94 replied:
Adicionado

brunodelucasbarbosa commented on src/components/FormSelect/index.jsx:
```diff
@@ -0,0 +1,44 @@
+      >
+        {value.elements.map((el) => (
+          <option key={el} value={el}>
+            {el[0].toUpperCase() + el.substring(1)}
```
Os valores do banco vem por padrão
`MESTRADO `ou `DOUTORADO` já

gabrielsoaresm94 replied:
Beleza, como criei um component para o formulário igual têm para o input, quis deixar o mais livre possível, para receber qualquer dado.

brunodelucasbarbosa commented on src/context/appContext.jsx:
```diff
@@ -102,6 +124,74 @@ function AppProvider({ children }) {
+        type: PRE_SAVE_USER_SUCCESS,
+        payload: { name_id, tax_id, email_id, phone_id, password }
+      })
+      saveUserToLocalStorage(name_id, tax_id, email_id, phone_id, password)
```
Depois que salvou no `LocalStorage` deve redirecionar o usuário para a tela de completar o cadastro

gabrielsoaresm94 replied:
Existe um método que faz isso, mas está falhando.

gabrielsoaresm94 replied:
Corrigido

brunodelucasbarbosa commented on src/context/appContext.jsx:
```diff
@@ -102,6 +124,74 @@ function AppProvider({ children }) {
+      if (!error?.response) {
+        dispatch({ type: SAVE_USER_ERROR, payload: 'Sem resposta do servidor' })
+      } else if (error?.response?.status === 400) {
+        dispatch({ type: SAVE_USER_ERROR, payload: 'Credenciais inválidas!' })
```
Esse credenciais inválidas é muito ruim para o usuário, não dá pra saber o que tá errado, se foi algum erro, se foi um ponto adicional, alguma coisa que ele esqueceu

gabrielsoaresm94 replied:
Acha melhor imprimir o erro do backend? Têm um aviso sobre o está faltando?

brunodelucasbarbosa replied:
Sim, o back-end valida as entradas do que está faltando, se algo não está no formato correto, vem no corpo da resposta

brunodelucasbarbosa commented:
![image](https://user-images.githubusercontent.com/60590932/194450388-72edba44-5ae3-49ca-a4ef-26fd58e26698.png)
Está aparecendo duas renderizações do toast

brunodelucasbarbosa commented:
![image](https://user-images.githubusercontent.com/60590932/194450958-88f7d7d5-db7f-48df-b8bb-32ccab0692b3.png)
`tax_id` está undefined
`advisor_id` está sem valor
`course` está sem valor

eu não sei o que aconteceu ai

gabrielsoaresm94 committed:
fix: corrige a maioria dos problemas das telas

lukedandrade committed:
Update README.md

Signed-off-by: Lucas de Andrade Lima <spirolucas@hotmail.com>

gabrielsoaresm94 committed:
fix: corrige e atualiza arquivos para integração

gabrielsoaresm94 committed:
Merge branch 'main' of https://github.com/SAB-UFBA2022/frontend into feature/cadastro

gabrielsoaresm94 committed:
Merge branch 'feature/cadastro' of https://github.com/SAB-UFBA2022/frontend into feature/cadastro

brunodelucasbarbosa commented on src/components/CadastroForm/index.jsx:
```diff
@@ -0,0 +1,128 @@
+    setValues({ ...values, [e.target.name]: e.target.value })
+  }
+
+  const handleSubmit = (e) => {
```
Na primeira parte do cadastro está avançando sem verificar nada....

gabrielsoaresm94 replied:
Foram adicionadas algumas verificações, a maioria dos inputs atuais têm. Vou marcar como resolvido.

brunodelucasbarbosa commented on src/components/CompleteCadastroForm/index.jsx:
```diff
@@ -0,0 +1,228 @@
+          label="Nome do orientador"
+          id="orientador_id"
+          name="orientador_id"
+          value={{
```
Ainda não tá puxando os orientadores do back

gabrielsoaresm94 replied:
Conversei com Lucas e como a proposta era entregar rápido o que faltava optei por deixar como hardcoded por enquanto. Como não é um erro específico, vou marcar como resolve.

brunodelucasbarbosa replied:
Rpz, a proposta de entregar rápido era pra meses atrás, mas, se não estiver quebrando...

brunodelucasbarbosa commented on src/App.js:
```diff
@@ -16,6 +18,8 @@ export default function App() {
       <Routes>
         {/* Rotas públicas */}
         <Route path="/" element={<Login />} />
+        <Route path="/cadastro" element={<Cadastro />} />
```
Pra ir até o cadastro, o cadastre-se não tá linkado!!

gabrielsoaresm94 replied:
Não entendi.

brunodelucasbarbosa replied:
Testa ai pra tu ver, desde o começo...
eu comentei isso antes

gabrielsoaresm94 replied:
Aparentemente já manda pra cadastro, se for a tela principal. Vou marcar como resolvido.

gabrielsoaresm94 replied:
Aparentemente foi corrigido, vou marcar como revolvido.

brunodelucasbarbosa commented on src/App.js:
```diff
@@ -16,6 +18,8 @@ export default function App() {
         {/* Rotas públicas */}
         <Route path="/" element={<Login />} />
+        <Route path="/cadastro" element={<Cadastro />} />
+        <Route path="/complete-cadastro" element={<CompleteCadastro />} />
```
Não tá validando nenhum campo, deu erro e ainda assim avançou, mostrando o erro na tela de login, e o usuário fica sem saber, até eu, n consigo saber ou fazer vários testes pra ver o q tá errado

gabrielsoaresm94 replied:
Adicionei validações, mas a maioria era com expressão regular, aí estavam quebrando. Existe alguma forma de utilizar isso no componente? Ou é preciso adicionar uma validação semelhante às que já existem no projeto?

brunodelucasbarbosa replied:
Que componente? não entendi.
Você pode usar qualquer validação, desde que funcione, sem precisar inventar, daí você trata os dados antes de enviar

brunodelucasbarbosa commented:
movido para draft até estar pronto

lukedandrade committed:
Correções de cadastro

Alterações de correção na página de cadastro e complete cadastro.
Agora, cadastro está validando os campos e informando quando o campo de entrada não é válido antes de prosseguir para complete-cadastro

lukedandrade committed:
Correções menores e Tentativa de leitura de resposta da requisição

Fiz alterações tentando fazer a leitura da resposta da requisição, temporariamente deixando a parte de navegação do completa-cadastro desabilitada para ficar na página e poder ver os resultados.

Problemas ainda existentes:
->Valores default dos select (Curso, Orientador, Agência) não são lidos, necessitando que usuário manualmente selecione.
->Resposta da requisição não está sendo mostrada em nenhum lugar, mesmo já estando configurada no backend.

gabrielsoaresm94 committed:
fix: remove import desnecessário e adiciona patternErro

gabrielsoaresm94 committed:
Merge branch 'main' of https://github.com/SAB-UFBA2022/frontend into feature/cadastro

gabrielsoaresm94 committed:
Merge branch 'main' of https://github.com/SAB-UFBA2022/frontend into feature/cadastro

gabrielsoaresm94 committed:
Merge branch 'main' of https://github.com/SAB-UFBA2022/frontend into feature/cadastro

gabrielsoaresm94 committed:
feat: add last pattern and change default values

gabrielsoaresm94 committed:
fix: last fix

brunodelucasbarbosa commented on src/components/CadastroForm/index.jsx:
```diff
@@ -0,0 +1,133 @@
+            autoComplete="off"
+            handleChange={handleChange}
+            pattern="[0-9]{10}"
+            patternErro="Telefone inválido, formato esperado: 9988888888"
```
Esse erro é mostrado várias vezes e não fica claro que formato é esse esperado....

brunodelucasbarbosa reviewed (changes requested):
Fica extremamente complicado revisar se eu não sei o que foi feito, são 11 arquivos editados no MR, dezenas de linhas alteradas, tem dias que eu estou vendo de pouco em pouco e por isso o atraso e o meu pedido para que outras pessoas pudessem revisar.
A segunda tela continua sem validação alguma pro usuário, inclusive já devo ter mencionado, mas, se um usuário cadastrar um cpf/taxid que já existe, não pode, cpfs/taxid são únicos, assim como outros
não consegui validar o caminho feliz
além de outras validações

lukedandrade committed:
Adição dos alertas

Finalmente, as responses são mostradas na tela do completa cadastro

lukedandrade committed:
Adição de Advisors no cadastro

Adição de método de chamada à API para conseguir os Advisors, juntamente com tentativas de popular o select através de métodos de repetição (Ainda sem sucesso). Pequenas correções feitas.

lukedandrade committed:
Mudanças pequenas para solucionar problemas envolvendo localStorage

Favor checar, essas alterações solucionam o problema de tentar um cadastro com alguns valores 'null' (Os oriundos da primeira parte do cadastro). Além disso, adicionei alterações para que ao voltar para a primeira parte do cadastro, o usuário mantém os dados previamente inseridos.

lukedandrade committed:
Forms de Select estão conectados à arrays agora.

Os SelectForm que viriam do banco de dados agora estão sendo populados por array. Funções como a getAdvisors() retornam os professores no banco no formato.

lukedandrade committed:
Adição de função para exclusão de Estudante (por ID)

Adição de função para exclusão de estudante. Chamada é feita de forma assíncrona, recebendo como parâmetro o ID do usuário a ser excluído. Feita com base na documentação do backend.

boliveira12 committed:
Fix nos problemas visuais da complete cadastro

boliveira12 committed:
Botao voltar no complete cadastro

gabrielsoaresm94 committed:
feat: update front

gabrielsoaresm94 committed:
feat: merge from last updates

lukedandrade committed:
Merge branch 'main' into feature/cadastro

Signed-off-by: Lucas de Andrade Lima <spirolucas@hotmail.com>

lukedandrade reviewed (approved):
Aprovado para funcionamento local.
